zk.el

```diff
@@ -278,19 +278,48 @@ Adds zk-id as an Embark target, and adds `zk-id-map' and
 
 (defun zk-file-p (&optional file strict)
   "Return t if FILE is a zk-file.
-If FILE is not given, get it from variable `buffer-file-name'. If
-STRICT is non-nil, make sure the file is in `zk-directory',
-otherwise just match against `zk-id-regexp'."
+If FILE is not given, get it from variable `buffer-file-name'.
+If STRICT is non-nil, make sure the file is in `zk-directory',
+otherwise just match against `zk-file-name-regexp'."
   (let ((file (cond ((stringp file) file)
                     ((null file) buffer-file-name)
                     ((listp file) (car file))
                     (t
                      (signal 'wrong-type-argument '(file))))))
     (and file
-         (file-exists-p file)
-         (string-match-p zk-id-regexp file)
+         (string-match (zk-file-name-regexp) file)
          (or (not strict)
-             (file-in-directory-p file zk-directory)))))
+             (save-match-data
+               (file-in-directory-p file zk-directory))))))
+
+(defun zk--de-separator (string)
```

Comment:
I don't think we need this function, since it doesn't save many lines, and it only obscures (abstracts) what is happening in the functions it appears it. Saving a few lines isn't a huge priority if it makes things less clear.